src/Microsoft.AspNetCore.Mvc.Core/UrlHelperExtensions.cs

```diff
@@ -201,6 +201,27 @@ public static string Action(this IUrlHelper helper, string action, string contro
             });
         }
 
+        /// <summary>
+        /// Generates an absolute URL for the specified <paramref name="action"/> and action
+        /// <paramref name="values"/>, which contains the protocol (such as "http" or "https") and host name from the
+        /// current request.
+        /// </summary>
+        /// <param name="helper">The <see cref="IUrlHelper"/>.</param>
+        /// <param name="action">The name of the action that is used to generate URL.</param>
+        /// <param name="values">An object that contains action values.</param>
+        /// <returns>The generated absolute URL.</returns>
+        public static string AbsoluteAction(this IUrlHelper helper, string action, object values)
```

Comment:
@mkArtakMSFT \ @Eilon thoughts on the name? Would it be better if this was an `bool generateAbsoluteUrl` for the existing set of overloads that do not accept a host \ scheme?

Reply:
What are the pros/cons of each approach?

Reply:
I'm not entirely sure if there's blockers, but if we change it to use `optional` arguments in 3.0, we could consistently add this option to all the `Url.*` options (`Url.Route`, `Url.Page` etc) without having an explosion in the number of extension methods.

Reply:
What about having a `Url.Absolute()` instead that would turn a relative URL into an absolute one?

So this would become `Url.Absolute(Url.Action(…))` but at the same time we also had `Url.Absolute(Url.Route(…))` or `Url.Absolute(Url.Page(…))` etc. without breaking compatibility and without ending up with lots of new methods.

And we could also use that implementation directly for #8279 then.

Reply:
@poke that sounds like a pretty neat idea. We're likely going to go with this approach.